src/action/common/place_nix_configuration.rs

```diff
@@ -28,65 +28,191 @@ impl PlaceNixConfiguration {
+                    if let (Some(setting_name), Some(setting_value)) = (setting_name, setting_value)
+                    {
+                        existing_conf_settings
+                            .push((setting_name.to_string(), setting_value.to_string()));
+                    }
+                }
+            }
+            tracing::trace!(
+                existing_conf_settings = existing_conf_settings
+                    .iter()
+                    .map(|(v, _)| v.to_string())
+                    .collect::<Vec<_>>()
+                    .join(","),
+                "Found existing config settings"
+            );
+
+            // Some settings (eg `experimental-features`) we must be able to set it.
+            let mut required_settings = vec!["experimental-features"];
+            #[cfg(not(target_os = "macos"))]
+            required_settings.push("auto-allocate-uids");
+            for required_setting in required_settings {
+                if let Some((existing_setting, existing_value)) = existing_conf_settings
+                    .iter()
+                    .find(|(k, _v)| k == required_setting)
+                {
+                    let required_setting = defaults_conf_settings
+                        .iter()
+                        .find(|(default_setting, _v)| default_setting.starts_with(existing_setting))
+                        .expect("Required setting was not planned -- please report this");
+                    if *existing_value != required_setting.1 {
+                        return Err(ActionErrorKind::ConfigurationConflict {
+                            setting: existing_setting.to_string(),
+                            existing_value: existing_value.to_string(),
+                            planned_value: required_setting.1.clone(),
+                        })
+                        .map_err(Self::error);
+                    } else {
+                        tracing::trace!(
+                            "Found existing setting `{} = {existing_value}` in config, continuing",
+                            required_setting.0
+                        )
+                    }
+                }
+            }
+
+            // Other settings, it's just a warning
+            for defaults_conf_setting in &defaults_conf_settings {
+                // We only set plain values so no need to be complicated.
+                for (existing_field, existing_value) in &existing_conf_settings {
+                    if defaults_conf_setting.0.trim() == *existing_field
+                        && defaults_conf_setting.1.trim() != existing_value.trim()
+                    {
+                        tracing::warn!("Found existing `/etc/nix/nix.conf` setting `{existing_field} = {existing_value}` which will override a default setting from the `nix-installer`, consider unsetting it. For settings like `experimental-features` you can use an `extra-*` prefix to append to the defaults")
```

Comment:
nit:

This isn't necessarily true if they used an `include` statement

Reply:
True but I didn't really want to implement full tracking on where we include various settings from. If you think it's important, let's do it, otherwise I just wanted to keep the code simpler. It's my impression most people will see an error like this and go "Oh, that's an old install" and simply blow away the `/etc/nix` folder.